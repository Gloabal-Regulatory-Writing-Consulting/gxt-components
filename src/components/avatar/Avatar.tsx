import React, { ChangeEvent, useMemo } from "react";
import { IconType } from "../svg/SvgIcon";
import {
  AvatarBox,
  AvatarButton,
  AvatarContainer,
  AvatarIcon,
  AvatarText,
  StyledIcon,
  UserIcon,
  UserInitials,
} from "./AvatarStyledComponents";

export type AvatarProps = {
  size: "small" | "large";
  onClickHandler?: () => void;
  handleImageUpload?: (e: ChangeEvent<HTMLInputElement>) => void;
  user: UserObj;
  isExpanded?: boolean;
  imageUrl?: string;
};

export type UserObj = {
  firstName: string;
  lastName: string;
  avatar?: string;
};

const Avatar: React.FC<AvatarProps> = ({
  size,
  onClickHandler,
  handleImageUpload,
  user,
  isExpanded,
  imageUrl,
}: AvatarProps) => {
  const { firstName, lastName, avatar } = user;
  const userName = `${firstName} ${lastName}`;

  const userInitials = useMemo(() => {
    const lastNameParts = lastName?.split(/[ -]/).filter((e) => e);
    const lastPart = lastNameParts?.at(-1) ?? "";
    return (firstName?.charAt(0) || "") + (lastPart?.charAt(0) || "");
  }, [firstName, lastName]);

  const noIcon: JSX.Element = (
    <UserInitials data-testid="user-initials" size={size}>{userInitials}</UserInitials>
  );

  const userIcon: JSX.Element = (
    <UserIcon src={imageUrl || avatar} alt="Avatar" />
  );

  const userAvatar = useMemo(() => {
    switch (true) {
        case !!imageUrl || !!avatar:
            return userIcon;
        default:
            return noIcon;
    }
}, [imageUrl, avatar, userIcon, noIcon]);

  const editImage = handleImageUpload ? (
    <>
      <StyledIcon
        iconType={IconType.PENCIL}
        size={size === "small" ? 8 : 18}
      ></StyledIcon>
      <input
        type="file"
        data-testid="user-image-input"
        style={{ display: "none" }}
        id="img"
        name="img"
        accept="image/*"
        onChange={handleImageUpload}
      />
    </>
  ) : null;

  return (
    <AvatarContainer className={`${isExpanded && "expanded-avatar"}`}>
      <AvatarIcon data-testid="avatarIcon" onClick={onClickHandler}>
        <AvatarBox>
          <AvatarButton size={size} className={`avatar-button`}>
            {userAvatar}
          </AvatarButton>
          {<label htmlFor="img"> {editImage} </label>}
        </AvatarBox>
        {isExpanded && (
          <div>
            <AvatarText weight="bold" size={size}>
              {userName}
            </AvatarText>
            <AvatarText weight="normal" size={size}>View profile</AvatarText>
          </div>
        )}
      </AvatarIcon>
    </AvatarContainer>
  );
};

export default Avatar;
