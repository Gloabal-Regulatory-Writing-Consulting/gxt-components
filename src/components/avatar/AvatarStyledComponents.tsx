import styled, { css } from "styled-components";
import { SvgIcon } from "../svg";

export const AvatarContainer = styled.div`
  display: flex;
  align-items: center;

  &.expanded-avatar {
    transition: width 0.3s ease;
  }
`;

export const AvatarIcon = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
  gap: 0.75rem;
`;

export const AvatarBox = styled.div`
  position: relative;
`;

export const AvatarButton = styled.div<{ size: "small" | "large" }>`
  background: var(--Semantic-Tokens-Fills-Component-Primary-primary-default, #177BA6);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
  overflow: clip;

  ${({ size }) =>
    size === "small" &&
    css`
      height: 32px;
      width: 32px;
      border-radius: 50%;
    `}

  ${({ size }) =>
    size === "large" &&
    css`
      height: 64px;
      width: 64px;
      border-radius: 50%;
    `}

    ${AvatarContainer}:hover & {
    border: 2px solid var(--Semantic-Tokens-Borders-Primary-border-primary-default, #177BA6);
  }
`;

export const AvatarText = styled.p<{
  weight?: string;
  size: "small" | "large";
}>`
  color: #000;
  margin: 0;

  ${({ weight }) =>
    weight === "bold" &&
    css`
      font-weight: 700;
    `}

  ${({ weight }) =>
    weight === "normal" &&
    css`
      font-weight: 400;
      ${AvatarContainer}:focus & {
        color: var(--Semantic-Tokens-Text-text-medium, #9CA3AF);
      }
    `}

  ${({ size }) =>
    size === "small" &&
    css`
    font-size: 1rem;
    line-height: 150%;
    `}

  ${({ size }) =>
    size === "large" &&
    css`
      font-size: 2rem;
      line-height: 150%;
    `}
`;

export const UserInitials = styled.div<{ size: "small" | "large" }>`
  color: var(--Semantic-Tokens-Text-text-light, #FFF);
  text-align: center;

  font-family: Mulish;
  font-style: normal;
  line-height: 150%;


  ${({ size }) =>
    size === "small" &&
    css`
      font-size: 0.75rem;
      font-weight: 400;
      letter-spacing: 0.014px;
    `}

  ${({ size }) =>
    size === "large" &&
    css`
      font-size: 1.25rem;
      font-weight: 700;
      letter-spacing: 0.024px;
    `}
`;

export const UserIcon = styled.img`
  height: 100%;
  width: 100%;
  object-fit: cover;
`;

export const StyledIcon = styled(SvgIcon)`
  padding: 0.25rem;
  position: absolute;
  bottom: -8px;
  right: ${({ size }) => (size === 8 ? "0" : "5px")};
  border-radius: 50%;
  background-color: lightgrey;
  cursor: pointer;

  &:hover {
    background-color: var(--primary-50, #1C99CE);;
    color: white;
  }
`;
