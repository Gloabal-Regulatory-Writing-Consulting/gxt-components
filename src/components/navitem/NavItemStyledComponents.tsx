import { NavLink } from "react-router-dom";
import styled, { css } from "styled-components";

export const NavItemContainer = styled.div<{ isLinkActive: boolean }>`
  display: flex;
  width: 15.75rem;
  padding: 0.5rem 0.75rem;
  align-items: center;
  gap: 0.75rem;

  ${({ isLinkActive }) =>
    isLinkActive === true &&
    css`
      border-left: solid;
      border-left-width: 0.25rem;
      border-left-color: var(--primary-200, #177ba6);
    `}
`;

export const NavLinkStyled = styled(NavLink)`
  text-decoration: none;
`;

export const NavIcon = styled.div`
  display: flex;
  width: 1.5625rem;
  height: 1.5625rem;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
`;

export const NavText = styled.p<{ isActive: boolean; isExpanded: boolean }>`
  font-size: 1rem;
  font-style: normal;
  font-weight: 700;
  line-height: 150%;
  letter-spacing: 0.00119rem;
  margin: 0;
  opacity: ${({ isExpanded }) => (isExpanded ? 1 : 0)};
  visibility: ${({ isExpanded }) => (isExpanded ? "visible" : "hidden")};
  transition:
    opacity 0.5s ease-in-out,
    visibility 0.5s ease-in-out;

  color: ${({ isActive }) =>
    isActive ? "var(--primary-200, #177BA6)" : "var(--neutral-400, #414141)"};
`;
