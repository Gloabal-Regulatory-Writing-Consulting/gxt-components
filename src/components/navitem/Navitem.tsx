import { FC } from "react";
import {
  NavIcon,
  NavItemContainer,
  NavLinkStyled,
  NavText,
} from "./NavItemStyledComponents";

export interface INavItem {
  navigateTo: string;
  text: string;
  Icon?: React.ComponentType<any>;
  className?: string;
  onClickHandler?: () => void;
}

export interface NavItemProps extends INavItem {
  isExpanded: boolean;
  isLinkActive: (path: string) => boolean;
}

export interface IClientStyling {
  thumbnail: string;
  logo: string;
}

const Navitem: FC<NavItemProps> = ({
  navigateTo,
  text,
  Icon,
  isExpanded,
  isLinkActive,
  onClickHandler,
}) => {
  const isLinkActivated = isLinkActive(navigateTo);

  return (
    <div onClick={onClickHandler} data-testid={text + "-navlink"}>
      <NavLinkStyled
        data-tooltip-id="navlink"
        data-tooltip-content={text}
        to={navigateTo}
        data-testid={text}
        end
      >
        <NavItemContainer $isLinkActive={isLinkActivated}>
          <NavIcon>
            {Icon && (
              <Icon
                stroke={
                  isLinkActivated ? "var(--primary-200, #177BA6)" : undefined
                }
                fill={
                  isLinkActivated ? "var(--primary-200, #177BA6)" : undefined
                }
              ></Icon>
            )}
          </NavIcon>
          <NavText $isExpanded={isExpanded} $isActive={isLinkActivated}>
            {text}
          </NavText>
        </NavItemContainer>
      </NavLinkStyled>
    </div>
  );
};

export default Navitem;
