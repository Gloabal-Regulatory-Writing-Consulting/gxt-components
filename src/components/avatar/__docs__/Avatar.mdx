import { Canvas, Meta, Controls } from "@storybook/blocks";
import * as Avatar from "./Avatar.stories.tsx";

<Meta of={Avatar} title="Avatar" />

# Avatar

Avatar component with different props.

#### Example

<Canvas of={Avatar.AvatarWithUpload} />

## Usage

```tsx
import Avatar from "sld-ui";

const Example = () => {
  const onClickHandler = () => {
    // Handle on click of avatar
  };

  const handleImageUpload = (e) => {
    // Handle image upload
  };

  return (
    <Avatar
      size="small"
      user={{ firstName: "John", lastName: "Doe", avatar: "avatar.jpg" }}
      isExpanded={false}
      imageUrl="image.jpg"
      onClickHandler={onClickHandler}
      handleImageUpload={handleImageUpload}
    />
  );
};

export default Example;
```

## Props
<Controls of={Avatar.AvatarWithUpload} />
