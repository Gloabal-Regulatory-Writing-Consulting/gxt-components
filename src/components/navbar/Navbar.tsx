import {
  useState,
  useEffect,
  Dispatch,
  SetStateAction,
  useRef,
  FC,
} from "react";
import Avatar, { UserObj } from "../avatar/Avatar";
import { Navitem } from "../navitem";
import { INavItem } from "../navitem/Navitem";
import {
  FooterContent,
  Header,
  HeaderContainer,
  LeadingContent,
  NavList,
  Sidebar,
} from "./NavbarStyledComponents";

export interface NavbarProps {
  footerLinks: INavItem[];
  contentLinks: INavItem[];
  setShowAvatarMenu: Dispatch<SetStateAction<boolean>>;
  showAvatarMenu: boolean;
  clientStyling: ClientStyling;
  user: UserObj;
  isLinkActive: (path: string) => boolean;
  expansionTime?: number;
}

export interface ClientStyling {
  thumbnail: string;
  logo: string;
}

const Navbar: FC<NavbarProps> = ({
  footerLinks,
  contentLinks,
  setShowAvatarMenu,
  showAvatarMenu,
  clientStyling,
  user,
  isLinkActive,
  expansionTime = 300,
}) => {
  const { thumbnail, logo } = clientStyling;

  const [isExpanded, setIsExpanded] = useState(false);
  const [isTimeOutExpanded, setIsTimeOutExpanded] = useState(false);

  const timeoutId = useRef<NodeJS.Timeout>();

  const handleMouseEnter = () => setIsTimeOutExpanded(true);

  const handleMouseLeave = () => setIsTimeOutExpanded(false);

  useEffect(() => {
    timeoutId.current = setTimeout(() => {
      setIsExpanded(isTimeOutExpanded);
    }, expansionTime);

    return () => {
      clearTimeout(timeoutId.current);
      timeoutId.current = undefined;
    };
  }, [isTimeOutExpanded]);

  const avatarIcon = (
    <span>
      <Avatar
        size="small"
        onClickHandler={() => setShowAvatarMenu(!showAvatarMenu)}
        user={user}
        isExpanded={isExpanded}
      />
    </span>
  );

  return (
    <Sidebar
      $isExpanded={isExpanded}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      data-testid="sidebar"
    >
      <LeadingContent $isExpanded={isExpanded}>
        <HeaderContainer $isExpanded={isExpanded}>
          <Header
            src={isExpanded ? logo : thumbnail}
            alt="company logo"
            data-testid="logo"
            $isExpanded={isExpanded}
          />
        </HeaderContainer>
        <NavList $isExpanded={isExpanded}>
          {contentLinks.map((link) => (
            <Navitem
              key={link.text}
              navigateTo={link.navigateTo}
              text={link.text}
              Icon={link.Icon}
              isExpanded={isExpanded}
              isLinkActive={isLinkActive}
              onClickHandler={link.onClickHandler}
            />
          ))}
        </NavList>
      </LeadingContent>
      <FooterContent>
        {footerLinks.map((link) => (
          <Navitem
            key={link.text}
            navigateTo={link.navigateTo}
            text={link.text}
            Icon={link.Icon}
            isExpanded={isExpanded}
            isLinkActive={isLinkActive}
            onClickHandler={link.onClickHandler}
          />
        ))}
        {avatarIcon}
      </FooterContent>
    </Sidebar>
  );
};

export default Navbar;
