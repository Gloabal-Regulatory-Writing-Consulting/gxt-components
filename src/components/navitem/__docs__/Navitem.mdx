import { Canvas, Meta, Controls } from "@storybook/blocks";
import * as Navitem from "./Navitem.stories.tsx";

<Meta of={Navitem} title="Navitem" />

# Navitem

## Example

<Canvas of={Navitem.NavitemComponent} />

## Usage

```tsx
import Navitem from "sld-ui";
import { ReactComponent as HomeIcon } from "./icons/HomeIcon.svg";

const Example = () => {
  const isLinkActive = (path) => {
    // return true if path matches the current path of application
  };
  return (
    <Navitem
      navigateTo="/home"
      text="Home"
      Icon={HomeIcon}
      isExpanded={true}
      isLinkActive={isLinkActive}
    />
  );
};

export default Example;
```

## Props
Navitem component with different props.
<Controls of={Navitem.NavitemComponent} />
