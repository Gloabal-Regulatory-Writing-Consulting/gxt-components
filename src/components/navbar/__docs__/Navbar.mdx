import { Canvas, Meta, Controls } from "@storybook/blocks";
import * as Navbar from "./Navbar.stories.tsx";

<Meta of={Navbar} title="Navbar" />

# Navbar

## Example

<Canvas of={Navbar.NavbarComponent} />

## Usage

```tsx
import Navbar from "sld-ui";
import { ReactComponent as HomeIcon } from "./icons/HomeIcon.svg";
import { ReactComponent as SettingsIcon } from "./icons/SettingsIcon.svg";

const Example = () => {
  const footerLinks = [
    {
      permission: "/settings",
      text: "Settings",
      Icon: SettingsIcon,
      isExpanded: false,
    },
    {
      permission: "/help",
      text: "Help",
      Icon: HomeIcon,
      isExpanded: false,
    },
  ];

  const contentLinks = [
    {
      permission: "/home",
      text: "Home",
      Icon: HomeIcon,
      isExpanded: false,
    },
    {
      permission: "/dashboard",
      text: "Dashboard",
      Icon: HomeIcon,
      isExpanded: false,
    },
  ];

  const clientStyling = {
    thumbnail: "http://example.com/thumbnail.jpg",
    logo: "http://example.com/logo.jpg",
  };

  const user = {
    firstName: "John",
    lastName: "Doe",
    avatar: "http://example.com/avatar.jpg",
  };

  const [showAvatarMenu, setShowAvatarMenu] = React.useState(false);

  return (
    <Navbar
      footerLinks={footerLinks}
      contentLinks={contentLinks}
      setShowAvatarMenu={setShowAvatarMenu}
      showAvatarMenu={showAvatarMenu}
      clientStyling={clientStyling}
      user={user}
    />
  );
};

export default Example;
```

## Props
Navbar component with different props.
<Controls of={Navbar.NavbarComponent} />
