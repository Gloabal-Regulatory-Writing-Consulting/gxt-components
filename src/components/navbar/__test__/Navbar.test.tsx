import React from "react";
import {
  screen,
  fireEvent,
  render,
  waitFor,
  act,
} from "@testing-library/react";
import Navbar, { NavbarProps } from "../Navbar";
import { UserObj } from "../../avatar/Avatar";
import { INavItem } from "../../navitem/Navitem";
import { vi, it, describe, expect } from "vitest";
import { MemoryRouter, Route, Routes } from "react-router-dom";

const renderWithRouter = (ui: React.ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return render(
    <MemoryRouter initialEntries={[route]}>
      <Routes>
        <Route path="*" element={ui} />
      </Routes>
    </MemoryRouter>,
  );
};

const footerLinks: INavItem[] = [
  {
    navigateTo: "user",
    text: "Footer Link 1",
    Icon: () => <span>FL1</span>,
  },
  {
    navigateTo: "user",
    text: "Footer Link 2",
    Icon: () => <span>FL2</span>,
  },
];

const contentLinks: INavItem[] = [
  {
    navigateTo: "user",
    text: "Content Link 1",
    Icon: () => <span>CL1</span>,
  },
  {
    navigateTo: "user",
    text: "Content Link 2",
    Icon: () => <span>CL2</span>,
  },
];

const user: UserObj = {
  firstName: "Test",
  lastName: "User",
  avatar: "http://example.com/avatar.jpg",
};

const clientStyling = {
  thumbnail: "http://example.com/thumbnail.jpg",
  logo: "http://example.com/logo.jpg",
};

const setShowAvatarMenu = vi.fn();

const isLinkActive = (path: string) => path === "user";

const defaultProps: NavbarProps = {
  footerLinks,
  contentLinks,
  setShowAvatarMenu,
  showAvatarMenu: false,
  clientStyling,
  user,
  isLinkActive,
};

describe("Navbar Component", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = renderWithRouter(<Navbar {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("expands on mouse enter", async () => {
    renderWithRouter(<Navbar {...defaultProps} />);
    const sidebar = screen.getByTestId("sidebar");

    act(() => {
      fireEvent.mouseEnter(sidebar);
    });
    await waitFor(() => {
      expect(sidebar).toHaveStyle("width: 15.7rem");
    });
  });

  it("collapses on mouse leave", async () => {
    renderWithRouter(<Navbar {...defaultProps} />);
    const sidebar = screen.getByTestId("sidebar");

    act(() => {
      fireEvent.mouseEnter(sidebar);
    });
    await waitFor(() => {
      expect(sidebar).toHaveStyle("width: 15.7rem");
    });
    fireEvent.mouseLeave(sidebar);

    await waitFor(() => {
      expect(sidebar).toHaveStyle("width: 5rem");
    });
  });

  it("toggles avatar menu on avatar click", () => {
    renderWithRouter(<Navbar {...defaultProps} />);
    const avatar = screen.getByRole("img", { name: /avatar/i });

    fireEvent.click(avatar);
    expect(setShowAvatarMenu).toHaveBeenCalledWith(true);
  });

  it("renders content links", () => {
    renderWithRouter(<Navbar {...defaultProps} />);
    contentLinks.forEach((link) => {
      expect(screen.getByText(link.text)).toBeInTheDocument();
    });
  });

  it("renders footer links", () => {
    renderWithRouter(<Navbar {...defaultProps} />);
    footerLinks.forEach((link) => {
      expect(screen.getByText(link.text)).toBeInTheDocument();
    });
  });

  it("renders logo or thumbnail based on expansion state", async () => {
    renderWithRouter(<Navbar {...defaultProps} />);
    const logo = screen.getByTestId("logo");

    act(() => {
      fireEvent.mouseEnter(screen.getByTestId("sidebar"));
    });
    await waitFor(() => {
      expect(logo).toHaveAttribute("src", clientStyling.logo);
    });

    act(() => {
      fireEvent.mouseLeave(screen.getByTestId("sidebar"));
    });
    await waitFor(() => {
      expect(logo).toHaveAttribute("src", clientStyling.thumbnail);
    });
  });
});
