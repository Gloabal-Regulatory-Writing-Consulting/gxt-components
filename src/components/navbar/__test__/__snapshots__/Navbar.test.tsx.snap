// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar Component > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-cPiJYC iuUwxc"
    data-testid="sidebar"
  >
    <div
      class="sc-jlZhRR jRXYUB"
    >
      <div
        class="sc-jEABJj dyXKlO"
      >
        <img
          alt="company logo"
          class="sc-cwHqhk kQdbWR"
          data-testid="logo"
          src="http://example.com/thumbnail.jpg"
        />
      </div>
      <div
        class="sc-dLNtp jQpGFx"
      >
        <a
          class="sc-dhKdPU lfdZxK"
          data-testid="Content Link 1"
          data-tooltip-content="Content Link 1"
          data-tooltip-id="navlink"
          href="/user"
        >
          <div
            class="sc-jXbVAB cVasOk"
          >
            <div
              class="sc-kpDprT swCGl"
            >
              <span>
                CL1
              </span>
            </div>
            <p
              class="sc-dAlxHm wuRAv"
            >
              Content Link 1
            </p>
          </div>
        </a>
        <a
          class="sc-dhKdPU lfdZxK"
          data-testid="Content Link 2"
          data-tooltip-content="Content Link 2"
          data-tooltip-id="navlink"
          href="/user"
        >
          <div
            class="sc-jXbVAB cVasOk"
          >
            <div
              class="sc-kpDprT swCGl"
            >
              <span>
                CL2
              </span>
            </div>
            <p
              class="sc-dAlxHm wuRAv"
            >
              Content Link 2
            </p>
          </div>
        </a>
      </div>
    </div>
    <div
      class="sc-eDPFhE gQNJRC"
    >
      <a
        class="sc-dhKdPU lfdZxK"
        data-testid="Footer Link 1"
        data-tooltip-content="Footer Link 1"
        data-tooltip-id="navlink"
        href="/user"
      >
        <div
          class="sc-jXbVAB cVasOk"
        >
          <div
            class="sc-kpDprT swCGl"
          >
            <span>
              FL1
            </span>
          </div>
          <p
            class="sc-dAlxHm wuRAv"
          >
            Footer Link 1
          </p>
        </div>
      </a>
      <a
        class="sc-dhKdPU lfdZxK"
        data-testid="Footer Link 2"
        data-tooltip-content="Footer Link 2"
        data-tooltip-id="navlink"
        href="/user"
      >
        <div
          class="sc-jXbVAB cVasOk"
        >
          <div
            class="sc-kpDprT swCGl"
          >
            <span>
              FL2
            </span>
          </div>
          <p
            class="sc-dAlxHm wuRAv"
          >
            Footer Link 2
          </p>
        </div>
      </a>
      <span>
        <div
          class="sc-aYaIB elxlrp false"
        >
          <div
            class="sc-gEvDqW KVfas"
            data-testid="avatarIcon"
          >
            <div
              class="sc-eqUzNf engioi"
            >
              <div
                class="sc-fqkwJk dnnfqZ avatar-button"
              >
                <img
                  alt="Avatar"
                  class="sc-gsFSjX hcoWMc"
                  src="http://example.com/avatar.jpg"
                />
              </div>
              <label
                for="img"
              >
                  
              </label>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
